apps/server/src/modules/files-storage/uc/files-storage.uc.spec.ts

```diff
@@ -450,277 +503,426 @@ describe('FilesStorageUC', () => {
 	});
 
-	describe('deleteFilesOfParent()', () => {
-		let requestParams: FileRecordParams;
-		beforeEach(() => {
-			requestParams = {
-				schoolId,
-				parentId: userId,
-				parentType: FileRecordParentType.User,
+	describe('deleteFilesOfParent is called', () => {
+		describe('WHEN user is authorized', () => {
+			const setup = () => {
+				const { requestParams1, userId1 } = getParams();
+
+				return { requestParams1, userId1 };
 			};
-			fileRecordRepo.findBySchoolIdAndParentId.mockResolvedValue([fileRecords, 1]);
-			storageClient.delete.mockResolvedValue([]);
-		});
 
-		describe('calls to fileRecordRepo.findBySchoolIdAndParentId()', () => {
-			it('should call once', async () => {
-				await service.deleteFilesOfParent(userId, requestParams);
-				expect(fileRecordRepo.findBySchoolIdAndParentId).toHaveBeenCalledTimes(1);
-			});
+			it('should call authorizationService.checkPermissionByReferences', async () => {
+				const { userId1, requestParams1 } = getParams();
+				const allowedType = FileStorageMapper.mapToAllowedAuthorizationEntityType(requestParams1.parentType);
 
-			it('should call with correctly params', async () => {
-				await service.deleteFilesOfParent(userId, requestParams);
-				expect(fileRecordRepo.findBySchoolIdAndParentId).toHaveBeenCalledWith(
-					requestParams.schoolId,
-					requestParams.parentId
+				await filesStorageUC.deleteFilesOfParent(userId1, requestParams1);
+				expect(authorizationService.checkPermissionByReferences).toBeCalledWith(
+					userId1,
+					allowedType,
+					requestParams1.parentId,
+					PermissionContexts.delete
 				);
 			});
 
-			it('should throw error if entity not found', async () => {
-				fileRecordRepo.findBySchoolIdAndParentId.mockRejectedValue(new Error());
-				await expect(service.deleteFilesOfParent(userId, requestParams)).rejects.toThrow();
+			it('should call service with correct params', async () => {
+				const { requestParams1, userId1 } = setup();
+				await filesStorageUC.deleteFilesOfParent(userId1, requestParams1);
+				expect(filesStorageService.deleteFilesOfParent).toHaveBeenCalledWith(requestParams1);
 			});
 		});
 
-		describe('calls to fileRecordRepo.save()', () => {
-			it('should call with correctly params', async () => {
-				await service.deleteFilesOfParent(userId, requestParams);
-				expect(fileRecordRepo.save).toHaveBeenCalledWith(fileRecords);
-			});
+		describe('WHEN user is not authorized', () => {
+			const setup = () => {
+				const { requestParams1, userId1 } = getParams();
+				authorizationService.checkPermissionByReferences.mockRejectedValueOnce(new ForbiddenException());
 
-			it('should throw error if entity not found', async () => {
-				fileRecordRepo.save.mockRejectedValue(new Error());
-				await expect(service.deleteFilesOfParent(userId, requestParams)).rejects.toThrow();
+				return { requestParams1, userId1 };
+			};
+
+			it('should throw forbidden error', async () => {
+				const { requestParams1, userId1 } = setup();
```

Comment:
empty lines between preparation - execution - testing in this and the next tests.